src/components/bookmark-button/index.tsx

```diff
@@ -1,35 +1,52 @@
-import { AUTH_ENABLED } from 'hooks/use-authentication'
+import { IconBookmarkAdd16 } from '@hashicorp/flight-icons/svg-react/bookmark-add-16'
+import { IconBookmarkRemove16 } from '@hashicorp/flight-icons/svg-react/bookmark-remove-16'
+import Button from 'components/button'
+import { withState } from './with-state'
 import { RemoveBookmarkIcon, AddBookmarkIcon } from './icons'
+import { BookmarkButtonConfigType, BookmarkButtonProps } from './types'
 import s from './bookmark-button.module.css'
 
-/**
- * TODO
- * - create a 'plugged' in HOC that can render modals, update bookmark state etc.
- * This HOC will be used by this button and the one in tutorial meta
- */
-
-interface BookmarkButtonProps {
-	isBookmarked: boolean
+const bookmarkButtonConfig: BookmarkButtonConfigType = {
+	add: {
+		text: 'Add bookmark',
+		baseIcon: <IconBookmarkAdd16 />,
+		iconWithHover: <AddBookmarkIcon />,
+	},
+	remove: {
+		text: 'Remove bookmark',
+		baseIcon: <IconBookmarkRemove16 />,
+		iconWithHover: <RemoveBookmarkIcon />,
+	},
 }
 
-export default function BookmarkButton({ isBookmarked }: BookmarkButtonProps) {
-	// NOTE! - hiding this component from prod until auth is enabled
-	if (!AUTH_ENABLED) {
-		return null
-	}
-	const helpText = isBookmarked ? `Remove bookmark` : `Add bookmark`
+function BookmarkButtonIconOnly({
+	isBookmarked,
+	handleOnClick, // this is configured by the `withState` hoc
+}: BookmarkButtonProps) {
+	const { add, remove } = bookmarkButtonConfig
+	const ariaLabel = isBookmarked ? remove.text : add.text
 	return (
 		<button
 			aria-pressed={isBookmarked}
-			onClick={() => {
-				// TODO use the create / destroy methods in the client
-				// or render dialog to prompt auth if not auth'd
-				console.log('Bookmark clicked!')
-			}}
-			aria-label={helpText}
+			onClick={handleOnClick}
+			aria-label={ariaLabel}
 			className={s.button}
 		>
-			{isBookmarked ? <RemoveBookmarkIcon /> : <AddBookmarkIcon />}
+			{isBookmarked ? remove.iconWithHover : add.iconWithHover}
 		</button>
 	)
 }
+
+function BookmarkButtonTextAndIcon({
```

Comment:
This component was previously defined inline in the `TutorialMeta` component. Moved it over here to link it up with the state / interaction mgmt and co-locate with components that serve a similar purpose.